src/main/scala/tilelink/Monitor.scala

```diff
@@ -517,6 +610,98 @@ class TLMonitor(args: TLMonitorArgs, monitorDir: MonitorDirection = MonitorDirec
     when (bundle.a.fire() || bundle.d.fire()) { watchdog := 0.U }
   }
 
+  def legalizeADSourceStrict(bundle: TLBundle, edge: TLEdge) {
+    val a_size_bus_size = edge.bundle.sizeBits + 1 //add one so that 0 is not mapped to anything (size 0 -> size 1 in map, size 0 in map means unset)
+    val a_opcode_bus_size = 3 + 1 //opcode size is 3, but add so that 0 is not mapped to anything
+    val log_a_opcode_bus_size = log2Ceil(a_opcode_bus_size)
+    val log_a_size_bus_size = log2Ceil(a_size_bus_size)
+    def size_to_numfullbits(x: UInt): UInt = (1.U << x) - 1.U //convert a number to that many full bits
+
```

Comment:
Since you're using `@chiselName`, can't you get rid of all of these `suggestNames`?

Reply:
For some reason @chiselName wasn't working for me